src/persist-client/src/batch.rs

```diff
@@ -168,17 +173,25 @@ where
     D: Semigroup + Codec64,
 {
     pub(crate) fn new(
+        cfg: PersistConfig,
         size_hint: usize,
         lower: Antichain<T>,
         blob: Arc<dyn BlobMulti + Send + Sync>,
         shard_id: ShardId,
     ) -> Self {
+        let parts = BatchParts::new(
+            cfg.batch_builder_max_outstanding_parts,
+            shard_id,
+            lower.clone(),
+            Arc::clone(&blob),
+        );
         Self {
             size_hint,
             lower,
             max_ts: T::minimum(),
-            records: ColumnarRecordsVecBuilder::default(),
+            records: ColumnarRecordsVecBuilder::new_with_len(cfg.blob_target_size),
```

Comment:
Before I started the review and saw this I was worried that we might now be producing too many small blobs! :raised_hands: 

I had started thinking that we might want to expose multi-part uploads from the `Blob` API, that way, we could thread pipelining all the way through to `Blob`. But it's not as straightforward as just tweezing apart `Blob::set()`.